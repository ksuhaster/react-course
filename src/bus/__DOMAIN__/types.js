export const types = Object.freeze({
    DOMAIN_ACTION: 'DOMAIN_ACTION'
});