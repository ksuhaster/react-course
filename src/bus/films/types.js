export const types = Object.freeze({

    // === PEOPLE ===

    // Sync
    FILMS_START_FETCHING: 'FILMS_START_FETCHING',
    FILMS_STOP_FETCHING: 'FILMS_STOP_FETCHING',
    FILMS_FILL: 'FILMS_FILL',
    FILMS_SET_FETCHING_ERROR: 'FILMS_SET_FETCHING_ERROR',

    // Async
    FILMS_FETCH_ASYNC: 'FILMS_FETCH_ASYNC',

    // === PERSON ===

    // Sync
    FILM_START_FETCHING: 'FILM_START_FETCHING',
    FILM_STOP_FETCHING: 'FILM_STOP_FETCHING',
    FILM_FILL: 'FILM_FILL',
    FILM_SET_FETCHING_ERROR: 'FILM_SET_FETCHING_ERROR',

    // Async
    FILM_FETCH_ASYNC: 'FILM_FETCH_ASYNC',
});
