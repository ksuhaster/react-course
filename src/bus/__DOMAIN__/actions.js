// Types
import { types } from './types';

export const domainActions = Object.freeze({
    domainAction: () => {
        return {
            type: types.DOMAIN_ACTION
        }
    }
});