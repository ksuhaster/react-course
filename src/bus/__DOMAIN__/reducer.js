// Types
import { types } from './types';

const initialState = {
    property: ''
};

export const domainReducer = ( state = initialState, { type } ) => {
    switch (type) {
        case types.DOMAIN_ACTION:
            return { ...state, property: '' };
    
        default:
            return state;
    }
}