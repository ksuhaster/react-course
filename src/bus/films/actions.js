// Types
import { types } from './types';

import { api } from '../../api';

export const filmsActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILMS_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILMS_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILMS_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILMS_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.FILMS_FETCH_ASYNC
        });

        dispatch(filmsActions.startFetching());

        const response = await api.films.fetch();

        if (response.status === 200) {
            const { results } = await response.json();
            dispatch(filmsActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmsActions.setFetchingError(error));
        }

        dispatch(filmsActions.stopFetching());
    }
});


export const filmActions = Object.freeze({
    // Sync
    startFetching: () => {
        return {
            type: types.FILM_START_FETCHING,
        }
    },
    stopFetching: () => {
        return {
            type: types.FILM_STOP_FETCHING,
        }
    },
    fill: (payload) => {
        return {
            type: types.FILM_FILL,
            payload
        }
    },
    setFetchingError: (error) => {
        return {
            type: types.FILM_SET_FETCHING_ERROR,
            error: true,
            payload: error
        }
    },
    // Async
    fetchAsync: (id) => async (dispatch) => {
        dispatch({
            type: types.FILM_FETCH_ASYNC
        });

        dispatch(filmActions.startFetching());

        const response = await api.film.fetch(id);

        if (response.status === 200) {
            const results = await response.json();
            dispatch(filmActions.fill(results));
        } else {
            const error = {
                status: response.status
            };

            dispatch(filmActions.setFetchingError(error));
        }

        dispatch(filmActions.stopFetching());
    }
});