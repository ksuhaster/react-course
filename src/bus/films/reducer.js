// Types
import { types } from './types';

const initialState = {
    data: [],
    isFetching: false,
    error: false
};

export const filmsReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.FILMS_START_FETCHING:
            return { ...state, isFetching: true };
        case types.FILMS_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.FILMS_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.FILMS_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
};

export const filmReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.FILM_START_FETCHING:
            return { ...state, isFetching: true };
        case types.FILM_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.FILM_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.FILM_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
};