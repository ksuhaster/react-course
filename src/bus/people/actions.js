// Types
import { types } from './types';

import { api } from '../../api';

export const peopleActions = Object.freeze({
    // Sync
    fill: (payload) => {
        return {
            type: types.PEOPLE_FILL,
            payload
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.PEOPLE_FETCH_ASYNC
        });

        const response = await api.people.fetch();
        const { results } = await response.json();

        dispatch(peopleActions.fill(results));
    }
});