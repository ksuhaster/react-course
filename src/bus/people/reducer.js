// Types
import { types } from './types';

const initialState = {
    data: [],
    isFetching: false,
    error: false
};

export const peopleReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.PEOPLE_START_FETCHING:
            return { ...state, isFetching: true };
        case types.PEOPLE_STOP_FETCHING:
            return { ...state, isFetching: false };
        case types.PEOPLE_SET_FETCHING_ERROR:
            return { ...state, error: payload };
        case types.PEOPLE_FILL:
            return { ...state, data: payload };

        default:
            return state;
    }
};