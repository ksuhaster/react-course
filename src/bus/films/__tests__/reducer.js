import { filmsReducer, filmReducer } from '../reducer';
import { filmsActions, filmActions } from '../actions';
import { types } from "../types";


describe('film list reducer:', () => {
    test('should return initial state by default', () => {
        expect(filmsReducer(void 0, {})).toMatchSnapshot();
    });

    // Sync
    test(`should handle ${types.FILMS_START_FETCHING} action`, () => {
        expect(filmsReducer(void 0, filmsActions.startFetching())).toMatchSnapshot();
    });
    test(`should handle ${types.FILMS_STOP_FETCHING} action`, () => {
        expect(filmsReducer(void 0, filmsActions.stopFetching())).toMatchSnapshot();
    });
    test(`should handle ${types.FILMS_SET_FETCHING_ERROR} action`, () => {
        expect(filmsReducer(void 0, filmsActions.setFetchingError())).toMatchSnapshot();
    });
    test(`should handle ${types.FILMS_FILL} action`, () => {
        expect(filmsReducer(void 0, filmsActions.fill())).toMatchSnapshot();
    });

    // Async
    test(`should handle ${types.FILMS_FETCH_ASYNC} action`, () => {
        expect(filmsReducer(void 0, filmsActions.fetchAsync())).toMatchSnapshot();
    });
});


describe('one film reducer:', () => {
    test('should return initial state by default', () => {
        expect(filmReducer(void 0, {})).toMatchSnapshot();
    });

    // Sync
    test(`should handle ${types.FILM_START_FETCHING} action`, () => {
        expect(filmReducer(void 0, filmActions.startFetching())).toMatchSnapshot();
    });
    test(`should handle ${types.FILM_STOP_FETCHING} action`, () => {
        expect(filmReducer(void 0, filmActions.stopFetching())).toMatchSnapshot();
    });
    test(`should handle ${types.FILM_SET_FETCHING_ERROR} action`, () => {
        expect(filmReducer(void 0, filmActions.setFetchingError())).toMatchSnapshot();
    });
    test(`should handle ${types.FILM_FILL} action`, () => {
        expect(filmReducer(void 0, filmActions.fill())).toMatchSnapshot();
    });

    // Async
    test(`should handle ${types.FILM_FETCH_ASYNC} action`, () => {
        expect(filmReducer(void 0, filmActions.fetchAsync())).toMatchSnapshot();
    });
});
