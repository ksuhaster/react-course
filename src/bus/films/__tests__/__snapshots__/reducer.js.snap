// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`film list reducer: should handle FILMS_FETCH_ASYNC action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`film list reducer: should handle FILMS_FILL action 1`] = `
Object {
  "data": undefined,
  "error": false,
  "isFetching": false,
}
`;

exports[`film list reducer: should handle FILMS_SET_FETCHING_ERROR action 1`] = `
Object {
  "data": Array [],
  "error": undefined,
  "isFetching": false,
}
`;

exports[`film list reducer: should handle FILMS_START_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": true,
}
`;

exports[`film list reducer: should handle FILMS_STOP_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`film list reducer: should return initial state by default 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`one film reducer: should handle FILM_FETCH_ASYNC action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`one film reducer: should handle FILM_FILL action 1`] = `
Object {
  "data": undefined,
  "error": false,
  "isFetching": false,
}
`;

exports[`one film reducer: should handle FILM_SET_FETCHING_ERROR action 1`] = `
Object {
  "data": Array [],
  "error": undefined,
  "isFetching": false,
}
`;

exports[`one film reducer: should handle FILM_START_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": true,
}
`;

exports[`one film reducer: should handle FILM_STOP_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;

exports[`one film reducer: should return initial state by default 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;
